faasmtools/build.py

```diff
@@ -91,6 +144,63 @@
 WASM_LDSHARED = " ".join([WASM_CC] + WASM_LDFLAGS_SHARED)
 WASM_LDXXSHARED = " ".join([WASM_CXX] + WASM_LDFLAGS_SHARED)
 
+# Lib order very important here
+# TODO: remove this when BLAS support is moved to a task
+WASM_BLAS_LIBS = [
+    "lapack",
+    "blas",
+    "f2c",
+]
+
+# ----------
+# Env. variables for different environments
+# ----------
+
+# Env. variables as a dictionary: prefix with FAASM_WASM or FAASM_NATIVE
+# depending on the build type variables target
+FAASM_BUILD_ENV_DICT = {
+    "CMAKE_ROOT": FAASM_CMAKE_ROOT,
+    "FAASM_NATIVE_INSTALL_DIR": FAASM_NATIVE_DIR,
+    "FAASM_WASM_CC": WASM_CC,
+    "FAASM_WASM_CXX": WASM_CXX,
+    "FAASM_WASM_AR": WASM_AR,
+    "FAASM_WASM_NM": WASM_NM,
+    "FAASM_WASM_RANLIB": WASM_RANLIB,
+    "FAASM_WASM_HOST_SHARED": WASM_HOST_SHARED,
+    "FAASM_WASM_HOST_STATIC": WASM_HOST_STATIC,
+    "FAASM_WASM_HOST_UNKNOWN": WASM_HOST_UNKNOWN,
+    "FAASM_WASM_INSTALL_DIR": WASM_TOOLCHAIN_BIN,
+    "FAASM_WASM_SYSROOT": WASM_SYSROOT,
+    "FAASM_WASM_CFLAGS": " ".join(WASM_CFLAGS),
+    "FAASM_WASM_CFLAGS_SHARED": " ".join(WASM_CFLAGS_SHARED),
+    "FAASM_WASM_EXE_LINKER_FLAGS": " ".join(WASM_EXE_LDFLAGS),
+    "FAASM_WASM_EXE_LINKER_FLAGS_SHARED": " ".join(WASM_EXE_LDFLAGS_SHARED),
+    "FAASM_WASM_SHARED_LINKER_FLAGS": " ".join(WASM_LDFLAGS_SHARED),
+    "FAASM_WASM_STATIC_LINKER_FLAGS": " ".join(WASM_LDFLAGS),
+    "FAASM_WASM_BLAS_LIBS": " ".join(WASM_BLAS_LIBS),
+}
+
+# Note that values in the environment dictionary can only be strings, bytes, or
+# Path-like objects, otherwise python's subprocess module complains
+FAASM_RUNTIME_ENV_DICT = {
+    "FAASM_WASM_BYTES_PER_PAGE": str(FAASM_WASM_BYTES_PER_PAGE),
+    "FAASM_WASM_CTORS_FUNC_NAME": FAASM_WASM_CTORS_FUNC_NAME,
+    "FAASM_WASM_MAX_MEMORY": str(FAASM_WASM_MAX_MEMORY),
+    "FAASM_WASM_STACK_SIZE": str(FAASM_WASM_STACK_SIZE),
+    "FAASM_WASM_ZYGOTE_FUNC_NAME": FAASM_WASM_ZYGOTE_FUNC_NAME,
+}
+
+
+def get_dict_as_cmake_vars(env_dict):
```

Comment:
In Faasm, we can pass the `FAASM_RUNTIME_ENV_DICT` variables as CMake CACHE variables, which is preferrable than env. variables. Unfortunately, CACHE variables are not always read/loaded properly in toolchain files so in `faasm/cpp` we revert to using env. variables.